### PR TITLE
REP-5214 Take 3 - Schema updated to not allow definition of an empty Via header prefix.

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

File: repose-aggregator/core/repose-core/src/test/scala/org/openrepose/nodeservice/response/ResponseHeaderServiceImplTest.scala [hunk 1]
```diff
@@ -51,50 +51,33 @@ class ResponseHeaderServiceImplTest extends FunSpec with BeforeAndAfterEach with
...
-      ("1.0", null, true),
-      ("1.1", "", true),
-      ("1.1", null, true)
-    ) foreach { case (protocol, prefix, includeVersion) =>
-      it(s"should set the Via header with protocol '$protocol', default prefix 'Repose' not '$prefix', and version '$version'") {
+      it(s"should set the Via header with protocol '$protocol', default prefix 'Repose' not 'NULL', and version '$version'") {
         when(request.getProtocol).thenReturn(s"HTTP/$protocol")
-        when(containerConfigurationService.getResponseVia).thenReturn(Optional.ofNullable(prefix))
-        when(containerConfigurationService.includeViaReposeVersion()).thenReturn(includeVersion)
+        when(containerConfigurationService.getResponseVia).thenReturn(Optional.ofNullable(null.asInstanceOf[String]))
```
Comment (dmnjohns) [inline repose-aggregator/core/repose-core/src/test/scala/org/openrepose/nodeservice/response/ResponseHeaderServiceImplTest.scala]: Couldn't you do: `Optional.ofNullable[String](null)`? Slightly shorter, and you aren't casting a null.

**Comment by dmnjohns** [comment]
I like it. Apart from the one comment, I think this is good to go. 👍

File: repose-aggregator/core/repose-core/src/test/scala/org/openrepose/nodeservice/response/ResponseHeaderServiceImplTest.scala (hunk 1 above)
Comment (wdschei) [inline repose-aggregator/core/repose-core/src/test/scala/org/openrepose/nodeservice/response/ResponseHeaderServiceImplTest.scala]: Updated.

**Comment by wdschei** [comment]
This was also rebased on to `REP-5214_ViaHeaderConfiguration` to pick up @Mario-Lopez 's changes.

**Pull request merged**